tests/kernels/moe/test_sonic_moe.py

```diff
@@ -0,0 +1,268 @@
+# SPDX-License-Identifier: Apache-2.0
+# SPDX-FileCopyrightText: Copyright contributors to the vLLM project
+"""
+Tests for Sonic MoE integration.
+
+Sonic MoE is a high-performance MoE implementation optimized for Hopper GPUs.
+These tests verify correct integration with vLLM's modular MoE infrastructure.
+
+Requires:
+- sonicmoe package installed
+- NVIDIA Hopper GPU (H100/H200)
+"""
+
+import pytest
+import torch
+
+from tests.kernels.utils import torch_experts
+from vllm.config import ParallelConfig, VllmConfig, set_current_vllm_config
+from vllm.platforms import current_platform
+
+# Check for Sonic MoE availability
+try:
+    from vllm.model_executor.layers.fused_moe.config import FusedMoEQuantConfig
+    from vllm.model_executor.layers.fused_moe.fused_moe import fused_topk
+    from vllm.model_executor.layers.fused_moe.modular_kernel import (
+        FusedMoEModularKernel,
+    )
+    from vllm.model_executor.layers.fused_moe.prepare_finalize import (
+        MoEPrepareAndFinalizeNoEP,
+    )
+    from vllm.model_executor.layers.fused_moe.sonic_moe import (
+        SonicMoEExperts,
+        is_sonic_moe_supported,
+        sonic_moe,
+    )
+
+    SONICMOE_AVAILABLE = True
+except ImportError:
+    SONICMOE_AVAILABLE = False
+
+
+def _is_hopper_gpu() -> bool:
+    """Check if running on Hopper GPU."""
+    if not torch.cuda.is_available():
+        return False
+    try:
+        major, _ = torch.cuda.get_device_capability()
+        return major >= 9
+    except Exception:
+        return False
```

Comment:
This should use current_platform's is_cuda and is_device_capability interface, also it is only supported on hopper at the moment so needs equality